components/Bounty/AwardBountyModal.tsx

```diff
@@ -526,66 +526,76 @@ function AwardBountyModal({
-                icon={["fal", "info-circle"]}
-                data-for={"distribute"}
-                data-tip
-              />
-            </div>
-          </div>
-          <div className={css(styles.userRows)}>
-            {threads?.map((thread) => {
-              const author = thread?.data?.created_by?.author_profile;
-              const mapKey = `${author.user}-${thread?.data.id}-award`;
-
-              return (
-                <div className={css(styles.awardUserRow)}>
-                  <AwardUserRow
-                    author={author}
-                    remainingAmount={bountyAmount}
-                    comment={thread}
-                    awardedAmount={userAwardMap[mapKey]}
-                    decreaseRemainingAmount={decreaseRemainingAmount}
-                    awardFullBounty={(e) => {
-                      e.stopPropagation();
-                      const userMap = {};
-
-                      userMap[mapKey] = bountyAmount;
-                      setUserAwardMap(userMap);
-                    }}
-                  />
-                </div>
-              );
-            })}
+            {threads?.length >= 1 && (
+              <div
+                className={css(styles.distribute)}
+                onClick={distributeOnUpvote}
+              >
+                <Image
+                  src="/shooting-star.png"
+                  alt="Distribute based on Upvote"
+                  className={css(styles.shootingStar)}
+                  width={15}
+                  height={14.93}
+                />{" "}
+                <span>Distribute based on Upvote</span>
+                <FontAwesomeIcon
+                  className={css(styles.info)}
+                  icon={["fal", "info-circle"]}
+                  data-for={"distribute"}
+                  data-tip
+                />
+              </div>
+            )}
           </div>
+
+          {threads?.length < 1 ? (
+            <div className={css(styles.noThreads)}>
+              No recipients to award yet.
+            </div>
+          ) : (
+            <div className={css(styles.userRows)}>
+              {threads?.map((thread) => {
+                const author = thread?.data?.created_by?.author_profile;
+                const mapKey = `${author.user}-${thread?.data.id}-award`;
+
+                return (
+                  <div className={css(styles.awardUserRow)}>
+                    <AwardUserRow
+                      author={author}
+                      remainingAmount={bountyAmount}
+                      comment={thread}
+                      awardedAmount={userAwardMap[mapKey]}
+                      decreaseRemainingAmount={decreaseRemainingAmount}
+                      awardFullBounty={(e) => {
+                        e.stopPropagation();
+                        const userMap = {};
+
+                        userMap[mapKey] = bountyAmount;
+                        setUserAwardMap(userMap);
```

Comment:
not sure what this is doing. but we can just inline this
```suggestion
                        const userMap = {};

                        userMap[mapKey] = bountyAmount;
                        setUserAwardMap({[mapKey]: bountyAmount});
```